### ISSUE TITLE
ImageArcGIS Layer misaligned 

### ISSUE THREAD

**Issue by tomasfreres** [openlayers/openlayers#10488]
Hi Folks,

I'm building a Webapp where I need to project a ImageArcGIS Layer on top of a regular map. Until today, I'm not able to reproduce this bug on my machine (Macbook Air 13" 2015).
The displayed ImageArcGIS Layer is being projected nowhere near it should be. At first, I thought the bug exists only on Chromium-based browsers, but recently I got notices that it also happens on Firefox. 

This is on Firefox
![grafik](https://user-images.githubusercontent.com/4959632/71841267-4bfd4e00-30bf-11ea-8bc2-8688f933d461.png)

This is on Chrome
![grafik](https://user-images.githubusercontent.com/4959632/71841301-60d9e180-30bf-11ea-9525-104410ac5850.png)

Weirdly, when used with Chromium installed on the same machine, the ArcGIS layer is projected as it should be, like this (Mozilla Firefox on my Machine):
![grafik](https://user-images.githubusercontent.com/4959632/71841355-7c44ec80-30bf-11ea-9bb5-55934d12b3e4.png)

My code to instantiate the ImageArcGIS Layer:
`overlayLayer = new ImageLayer({
        source: new ImageArcGISRest({
          url: layer.URL,
          params: layer.Parameters,
          crossOrigin: "anonymous"
        }),
        opacity: layer.Opacity,
        visible: true
      })`

It happens to all my inserted ArcGIS Layer, regardless of the service URL.

Until today I couldn't reproduce it. Either it's there, or not.

**Expected behavior**
The ArcGIS layer should be where it's supposed to be projected.

Thanks in advance,

Tomás

**Comment by tomasfreres** [comment]
I have narrowed the problem down. It has something to do with scaling. If you change the browser's scaling (to e.g. 30%), the layers are misaligned.

**Comment by ahocevar** [comment]
Can you reproduce that with one of the official examples? Or can you create an example on codesandbox.io that shows the issue?

**Comment by JamesxuWang** [comment]
i got this problom too. if i use chrome devtool , toggle device toolbar and set the resolution to 3784*1400, the img is not in the right position!

**Comment by aolagers** [comment]
I noticed the same problem with ol v6.3.1 on Firefox 76.0.1. For me it happens on a VectorLayer, so it does not seem to be specific to ArcGIS layers.

When I check the element with devtools, the canvas element element for the layer has its style set to: `"position: absolute; left: 0px; transform-origin: left top 0px; transform: matrix(0.8, 0, 0, 0.8, 0, 0);"`

The transform matrix scales the layer down to 80% which causes the layer to be misaligned with the basemap.

Does not happen on Chrome 81.0.4044.

**Comment by pborissow** [comment]
This issue is very much alive and well even in v6.5.0. I was able to replicate the issue on Windows with Chome, Firefox, and Edge  the browser set >100% zoom and on a MacBook Air with a Retina display at 100% zoom. Something to do with the pixel density. 

@aolagers comment above with the transform matrix offered me a clue of how to implement a workaround (aka hack). Basically, I watch for resize and extent change events and update the style attributes for all the canvas elements associated with a view port (1 per layer I believe).

`      
        var arr = viewport.getElementsByTagName("canvas");
        for (var i=0; i<arr.length; i++){
            var canvas = arr[i];
            var width = canvas.parentNode.offsetWidth;
            var height = canvas.parentNode.offsetHeight;
            canvas.style.width = width + "px";
            canvas.style.height = height + "px";
            if (disableTransform && canvas.style.transform){
                if (canvas.style.transform.length>0) canvas.style.transform = "";
            }
        }
` 

Interactive drawings via ol.interaction.Draw and ol.interaction.DragBox also get messed up with the transform. Ideally, we should have a way to programmatically disable the transform. I don't think I need it when showing a basic 2d map (north up, not rotated, not 3d, etc).

**Comment by ahocevar** [comment]
https://openlayers.org/en/v6.5.0/examples/arcgis-image.html uses an `ImageArcGISRest` layer and works fine. Any chance to get a reproducible example?

**Comment by pborissow** [comment]
@ahocevar I think I know what's going on. In the 5.x series of OL there was only 1 canvas per map. The width and height styles were set to 100% and it didn't resize properly in my app. So I wrote some code to update the width/height style attributes the canvas on resize as a workaround. 

I recently upgraded to 6.x and left my resize code in place. I just realized that in the 6.x series the width and height styles for the canvas are no longer set to 100% making my 5.x resize fix obsolete and actually issues. 

Bottom line, OL 6.5.0 is performing correctly. Sorry for the trouble.
